src/ecrc-frontend/src/components/page/consent/Consent.js

```diff
@@ -228,200 +222,41 @@ export default function Consent({
-          `/ecrc/private/getNextInvoiceId?orgTicketNumber=${orgTicketNumber}&requestGuid=${uuid}`,
-          {
-            headers: {
-              Authorization: `Bearer ${token}`
-            }
-          }
-        );
-      })
-      .then(getInvoiceResponse => {
-        if (!getInvoiceResponse) {
-          return getInvoiceResponse;
-        }
-
-        invoiceId = getInvoiceResponse.data.invoiceId;
-
-        appInfo = {
-          ...appInfo,
-          invoiceId
-        };
-
-        CRC.invoiceId = invoiceId;
-
-        return axios.post("/ecrc/private/createNewCRCService", CRC, {
-          headers: {
-            Authorization: `Bearer ${token}`
-          }
-        });
-      })
-      .then(newCRCResponse => {
-        if (!newCRCResponse) {
-          return newCRCResponse;
-        }
-        serviceId = newCRCResponse.data.serviceId;
-
-        appInfo = {
-          ...appInfo,
-          serviceId
-        };
-
-        if (orgApplicantRelationship === "ONETIME") {
-          setApplicationInfo(appInfo);
-          setLoading(false);
-          history.push("/criminalrecordcheck/success");
-        }
-
-        return axios.get(
-          `/ecrc/private/getServiceFeeAmount?orgTicketNumber=${orgTicketNumber}&scheduleTypeCd=${defaultScheduleTypeCd}&scopeLevelCd=${defaultCrcScopeLevelCd}&requestGuid=${uuid}`,
-          {
-            headers: {
-              Authorization: `Bearer ${token}`
-            }
-          }
-        );
-      })
-      .then(serviceFeeResponse => {
-        if (!serviceFeeResponse) {
-          return serviceFeeResponse;
-        }
-        serviceFeeAmount = serviceFeeResponse.data.serviceFeeAmount;
+      .then(singleCallResponse => {
+        const appResponseInfo = singleCallResponse.data;
 
         appInfo = {
           ...appInfo,
-          serviceFeeAmount
+          ...appResponseInfo
         };
 
         setApplicationInfo(appInfo);
         saveApplicant();
         saveOrg();
         saveApplicationInfo(appInfo);
 
-        const createURL = {
-          invoiceNumber: invoiceId,
-          requestGuid: uuid,
-          approvedPage: `${process.env.REACT_APP_FRONTEND_BASE_URL}/criminalrecordcheck/success`,
-          declinedPage: `${process.env.REACT_APP_FRONTEND_BASE_URL}/criminalrecordcheck/success`,
-          errorPage: `${process.env.REACT_APP_FRONTEND_BASE_URL}/criminalrecordcheck/success`,
-          totalItemsAmount: serviceFeeAmount,
-          serviceIdRef1: serviceId,
-          partyIdRef2: partyId
-        };
-
-        return axios.post("/ecrc/private/createPaymentUrl", createURL, {
-          headers: {
-            Authorization: `Bearer ${token}`
-          }
-        });
-      })
-      .then(urlResponse => {
-        if (!urlResponse) {
-          return;
+        if (appInfo.paymentUrl) {
+          window.location.href = appInfo.paymentUrl;
+          setLoading(false);
+        } else {
+          history.push("/criminalrecordcheck/success");
```

Comment:
I have a feeling this section of code will be caught as untested in sonarscan for test coverage
can we add a test case for when `appInfo` has `paymentUrl` and not? (unless I am mistaken and those tests exist in which case can ignore this comment)

Reply:
This should be checked with current tests, as it replaces the logic checking for VOLUNTEER vs EMPLOYEE. Can change to check if orgApplicantRelationship === EMPLOYEE if you think that would be more clear?

Reply:
lets push this out and see what lines show up as uncovered and add tests accordingly